cmd/sops/common/common.go

```diff
@@ -441,3 +441,19 @@ func PrettyPrintDiffs(diffs []Diff) {
 		}
 	}
 }
+
+func ReadFile(path string) (content []byte, err error) {
```

Comment:
I think this tries to do two things which really should be separate things.

In my opinion, handling of files should really be _just_ handling of files, with `os.Stdin` not classifying as such. Part because from the perspective of SOPS as an SDK, `decrypt.Decrypt` magically reading from `os.Stdin` would be kind of surprising.

Having said this, I would deal with `-` as a special path case within commands, and e.g. create another wrapper around `DataWithFormat` for an `io.Reader`.

Reply:
> files should really be just handling of files

I don't know. On Unix, stdin is just an other file descriptor and could be handle as file, too. (like `/dev/stdin`)

Living the unix philosophy, there should not be a difference between `/mnt/file` and `/proc/self/fd/1`.

From technical point of view, it may should fine. 

Creating an other wrapper, it possible. But the other wrapper would be the exact same copy/paste code.

Reply:
How is this the same copy/paste code?

```go
func Reader(r io.Reader, format string) (cleartext []byte, err error) {
        encryptedData, err := io.ReadAll(r)
	if err != nil {
		return nil, err
	}
	return DataWithFormat(encryptedData, FormatFromString(format))
}

Reply:
NB: I am fine with a function called `ReadFileOrStdin` (or something something) existing (and utilized) within `cmd/sops`, but I wouldn't want this to be the default for `decrypt`.